wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java

```diff
@@ -139,14 +139,16 @@ protected boolean isPublicRepository(T remoteFactoryUrl) {
     return false;
   }
 
-  private String formatUrl(String fileURL) throws DevfileException {
+  protected String formatUrl(String fileURL) throws DevfileException {
     String requestURL;
     try {
       if (new URI(fileURL).isAbsolute()) {
         requestURL = fileURL;
       } else {
-        // since files retrieved via REST, we cannot use path symbols like . ./ so cut them off
-        requestURL = remoteFactoryUrl.rawFileLocation(fileURL.replaceAll("^[/.]+", ""));
+        // since files retrieved via REST, we cannot use path like '.' or one that starts with './'
+        // or '../', so cut them off
```

Comment:
If we cut `../` of then we will have a wrong location, right?

Reply:
We addressing the resource, beginning from the repository root, and this cutting does not change anything.
But yes, it looks a bit confusing, just take a look at the test https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-factory-bitbucket-server/src/test/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketServerAuthorizingFileContentProviderTest.java#L115

Let me try to remove cutting of `../`, update the PR and republish `che-server` image, that is used to test.